applications/OptimizationApplication/python_scripts/algorithms/algorithm_steepest_descent.py

```diff
@@ -88,8 +89,12 @@ def ComputeSearchDirection(self, obj_grad) -> KratosOA.CollectiveExpression:
         return search_direction
 
     @time_decorator()
-    def ComputeControlUpdate(self, alpha) -> KratosOA.CollectiveExpression:
-        update = self.algorithm_data.GetBufferedData()["search_direction"] * alpha
+    def ComputeControlUpdate(self, alpha):
+        search_direction = self.algorithm_data.GetBufferedData()["search_direction"]
+        if type(alpha) is float:
+            update = search_direction * alpha
+        elif type(alpha) is KratosOA.CollectiveExpression:
+            update = search_direction.Scale(alpha)
```

Comment:
it might make sense to define an overload of `Scale` that takes a scalar argument (identical to `operator*(double)`) to avoid branches like this.

What do you think @sunethwarna ?